x-pack/plugin/blob-cache/src/main/java/org/elasticsearch/blobcache/shared/SharedBytes.java

```diff
@@ -215,27 +221,11 @@ protected void closeInternal() {
         }
     }
 
-    private final Map<Integer, IO> ios = ConcurrentCollections.newConcurrentMap();
-
     public IO getFileChannel(int sharedBytesPos) {
         assert fileChannel != null;
-        return ios.compute(sharedBytesPos, (p, io) -> {
-            if (io == null || io.tryIncRef() == false) {
-                final IO newIO;
-                boolean success = false;
-                incRef();
-                try {
-                    newIO = new IO(p);
-                    success = true;
-                } finally {
-                    if (success == false) {
-                        decRef();
-                    }
-                }
-                return newIO;
-            }
-            return io;
-        });
+        var res = ios[sharedBytesPos];
+        incRef();
+        return res;
```

Comment:
nit: can we avoid the `res` variable?

```suggestion
        incRef();
        return ios[sharedBytesPos];
```

Reply:
I'd rather not, if I ever have to debug a leak in this thing, I'll waste time hunting down whether or not I ever got an out of bounds exception here :)

Reply:
ah, makes sense, thanks.